### DISCUSSION TITLE
using watt as unit of heat source

### DISCUSSION THREAD

**Discussion by PJohnYu** [idaholab/moose#19670] [Q&A General] [answered]
Hello, I have a mesh with multi-blocks, and I want to use watt density as heat source at specified block. But I don't find ways to realize it. Can you please tell me some ideas?
Thank you

**Comment by GiudGiud** [comment]
Hello

You want to specify a heat source term in Watts?
If your mesh is in meters, and the rest of your simulation is in SI units, then you dont need to do any conversion. Watt is J/s, the SI unit. 

You can use the HeatSource
https://mooseframework.inl.gov/source/kernels/HeatSource.html
or the BodyForce kernels
https://mooseframework.inl.gov/source/kernels/BodyForce.html
to add a source term to an energy equation

Guillaume

**Reply by PJohnYu** [reply]
Thanks for your reply, Guillaume! 
I did a incomplete describe about my problem above. 
In my app, heat source is created by SolutionAux so it is stored in a AuxVariable. 
It maybe my  mistake that HeatSource and BodyForce can't seem to use AuxVariable object as source term. Do you have some suggestion?

**Reply by loganharbour** [reply]
You can scale your SolutionAux with the “scale_factor” parameter, where here you would use the conversion from your units -> W

**Comment by GiudGiud** [comment, marked as answer]
You can use the CoupledForce kernel to use an auxvariable as a heat source
https://mooseframework.inl.gov/source/kernels/CoupledForce.html

**Comment by PJohnYu** [comment]
Thank you very much. I have found the suitable cases following your suggestion.
